backends/rabbitmq-streams/read.go

```diff
@@ -0,0 +1,106 @@
+package rabbitmqStreams
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/batchcorp/plumber/printer"
+
+	"github.com/batchcorp/plumber/reader"
+
+	"github.com/jhump/protoreflect/desc"
+	"github.com/pkg/errors"
+	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/amqp"
+	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/stream"
+	"github.com/sirupsen/logrus"
+
+	"github.com/batchcorp/plumber/cli"
+)
+
+func Read(opts *cli.Options, md *desc.MessageDescriptor) error {
+	client, err := NewClient(opts)
+	if err != nil {
+		return err
+	}
+
+	r := &RabbitMQStreams{
+		Client:  client,
+		Options: opts,
+		MsgDesc: md,
+		log:     logrus.WithField("pkg", "rabbitmq-streams/read.go"),
+	}
+
+	return r.Read()
+}
+
+func (r *RabbitMQStreams) Read() error {
+	var count int
+
+	offsetOption, err := r.getOffsetOption()
+	if err != nil {
+		return errors.Wrap(err, "could not read messages")
+	}
+
+	handleMessage := func(consumerContext stream.ConsumerContext, message *amqp.Message) {
+		for _, value := range message.Data {
+			count++
+			data, err := reader.Decode(r.Options, r.MsgDesc, value)
+			if err != nil {
+				r.log.Error(err)
+				continue
+			}
+
+			printer.PrintRabbitMQStreamsResult(r.Options, count, consumerContext, message, data)
+		}
+
+		if !r.Options.ReadFollow {
+			consumerContext.Consumer.CloseHandler <- stream.Event{}
+			consumerContext.Consumer.Close()
```

Comment:
I think that this is not going to work., I need to check it
Why do you want to close the consumer inside the handler?

Reply:
I opened an issue https://github.com/rabbitmq/rabbitmq-stream-go-client/issues/57

Reply:
Plumber's default behavior is to read a single message and then exit, unless the user specifies the `--follow` flag. Since the handler is async, it wouldn't exit in time if I used my own close channel, so I had to close inside the handler, otherwise multiple messages would get read

Reply:
No problem, here the wip https://github.com/rabbitmq/rabbitmq-stream-go-client/pull/58

Reply:
I am going to remove:
```
consumerContext.Consumer.CloseHandler <- stream.Event{}
```
`CloseHandler` is not meant to be public

Reply:
Sounds good, thanks for the heads up!

Reply:
Please check the new version: https://github.com/rabbitmq/rabbitmq-stream-go-client/releases/tag/v0.10-alpha 
It contains the fix for the close issue and some input validation

Reply:
Perfect, thank you!